### PR TITLE
Enhance Permissions and Refactor Views for Quiz Sessions and Recordings

### PR DESCRIPTION
**Pull request by AustinfHunter**
This pull request introduces several enhancements to permission handling in the application, particularly focusing on quiz sessions and recordings. Additionally, it refactors code in relevant views to utilize the newly created permissions and streamline the permission checks. The changes are as follows:

- **Introduced `IsSessionOwner` Permission**:
  - Created a new permission class `IsSessionOwner` which allows only the owner of a quiz session to access relevant endpoints. This enhances security and access control.
  - The permission checks if the authenticated user is the instructor associated with the quiz session using the session code from URL parameters.

- **Refactored Views for Session Management**:
  - Updated `QuizSessionView`, `QuizSessionStudentInstructorView`, `QuizSessionInstructorView`, and `QuizSessionsByInstructorView` to use `IsSessionOwner` and `IsQuizOwner` as needed for appropriate access control.
  - Converted hardcoded permission checks into the new permission class to avoid redundancy and enhance code maintainability.

- **Updated Permissions for Recordings**:
  - Applied the `IsRecordingOwner` permission to various recording-related views, ensuring that users can only access or modify recordings they own.
  - Updated views such as `UpdateRecordingTitleView`, `UpdateTranscriptView`, `DeleteRecordingView`, and `GetTranscriptView` to enforce ownership directly through their permission classes.

- **Updated Test Cases**:
  - Adjusted the assertions in `test_change_recording_title_by_non_owner` to provide a clearer error response when permission checks fail, ensuring consistency across the API responses.

- **Code Refactor and Cleanup**:
  - Removed redundant inline permission checks within views that have been replaced with class-based permissions, improving clarity and reducing potential errors.
  - Revised import statements to maintain clean and organized code structure across different views.
  - General syntax and code quality improvements for better compliance with best practices.

This set of changes aims to create a more secure and maintainable code base by enforcing permissions more systematically, simplifying the handling of access control in views.

**Examples**
All examples are attempts at modifying another users resources.

Update quiz title
403-update-quiz-title

Update quiz settings
403-update-quiz-settings

Access quiz settings
403-settings

Access next question for session
403-next-question

Access quiz results
403-quiz-results

Access quiz session responses
403-quiz-session-responses

Delete session
403-delete-session

Access session
403-quiz-sessions

Update transcript
403-update-transcript

Get transcript
403-get-transcript

### CODE REVIEW

File: api/views/recordings_views.py [hunk 1]
```diff
@@ -73,7 +75,7 @@ def post(self, request):
 
 @extend_schema(tags=["Recordings"])
 class UpdateRecordingTitleView(APIView):
-    permission_classes = [IsAuthenticated]
+    permission_classes = [IsAuthenticated, IsRecordingOwner]
```
Comment (amhajja) [inline api/views/recordings_views.py:78]: Is `IsAuthenticated` redundant here?

File: api/views/session_views.py [hunk 2]
```diff
@@ -1,7 +1,15 @@
...
-from api.models import *
+from api.models import (
+    Quiz,
+    QuizSession,
+    QuizSessionStudent,
+    UserResponse,
+    QuestionMultipleChoice,
+    InstructorRecordings,
+    LectureSummary,
+)
```
Comment (amhajja) [inline api/views/session_views.py:12]: Was this flagged by `flake8`?

**Review by amhajja — CHANGES_REQUESTED** [review]
One request and one question.

File: api/views/recordings_views.py (hunk 1 above)
Comment (AustinfHunter) [reply api/views/recordings_views.py:78]: You're correct, it is redundant. I'll fix that correct that.

File: api/views/session_views.py (hunk 2 above)
Comment (AustinfHunter) [reply api/views/session_views.py:12]: No, but the star import was causing a lot of warnings in my editor. I don't know if PyCharm uses a different linter/language server than what I'm using, so maybe that was only an issue for me. I believe avoiding star imports is generally considered a best practice.

**Review by amhajja — APPROVED** [review]
Looks great!

**Pull request merged**